funfact/lang/_tsrex.py

```diff
@@ -130,16 +130,24 @@ def tensor(*spec, initializer=None):
         A tensor expression representing a single tensor object.
     '''
     if len(spec) == 2 and isinstance(spec[0], str) and _is_tensor(spec[1]):
-        raise NotImplementedError()
-    elif len(spec) == 1 and _is_tensor(spec[1]):
-        raise NotImplementedError()
+        symbol = spec[0]
+        initializer = spec[1]
+        size = initializer.shape
+    elif len(spec) == 1 and _is_tensor(spec[0]):
+        symbol = f'Anonymous_{AbstractTensor.n_nameless}'
+        initializer = spec[0]
+        size = initializer.shape
     elif isinstance(spec[0], str):
         symbol, *size = spec
     else:
         symbol = f'Anonymous_{AbstractTensor.n_nameless}'
         AbstractTensor.n_nameless += 1
         size = spec
 
+    if hasattr(size[0], "__len__"):
+        initializer = size[0]
```

Comment:
I'm a bit confused as to why `size[0]` could serve as the initializer.

Reply:
I have removed this, was from a previous version and should have been removed after rebasing